lib/snippets.js

```diff
@@ -0,0 +1,666 @@
+      this.subscriptions.add(watchDisposable)
+    })
+
+    this.subscriptions.add(atom.config.onDidChange('core.packagesWithSnippetsDisabled', ({newValue, oldValue}) => {
+       this.handleDisabledPackagesDidChange(newValue, oldValue)
+    }))
+
+    const snippets = this
+
+    this.subscriptions.add(atom.commands.add('atom-text-editor', {
+      'snippets:expand'(event) {
+        const editor = this.getModel()
+        if (snippets.snippetToExpandUnderCursor(editor)) {
+          snippets.clearExpansions(editor)
+          snippets.expandSnippetsUnderCursors(editor)
+        } else {
+          event.abortKeyBinding()
+        }
+      },
+
+      'snippets:next-tab-stop'(event) {
+        const editor = this.getModel()
+        if (!snippets.goToNextTabStop(editor)) { event.abortKeyBinding() }
+      },
+
+      'snippets:previous-tab-stop'(event) {
+        const editor = this.getModel()
+        if (!snippets.goToPreviousTabStop(editor)) { event.abortKeyBinding() }
+      },
+
+      'snippets:available'(event) {
+        const editor = this.getModel()
+        const SnippetsAvailable = require('./snippets-available')
+        if (snippets.availableSnippetsView == null) { snippets.availableSnippetsView = new SnippetsAvailable(snippets) }
+        snippets.availableSnippetsView.toggle(editor)
+      }
+    }))
+  },
+
+  deactivate () {
+    if (this.emitter != null) {
+      this.emitter.dispose()
+    }
+    this.emitter = null
+    this.editorSnippetExpansions = null
+    atom.config.transact(() => this.subscriptions.dispose())
+  },
+
+  getUserSnippetsPath () {
+    if (this.userSnippetsPath != null) { return this.userSnippetsPath }
+
+    this.userSnippetsPath = CSON.resolve(path.join(atom.getConfigDirPath(), 'snippets'))
+    if (this.userSnippetsPath == null) { this.userSnippetsPath = path.join(atom.getConfigDirPath(), 'snippets.cson') }
+    return this.userSnippetsPath
+  },
+
+  loadAll () {
+    this.loadBundledSnippets(bundledSnippets => {
+      this.loadPackageSnippets(packageSnippets => {
+        this.loadUserSnippets(userSnippets => {
+          atom.config.transact(() => {
+            for (const snippetSet of [bundledSnippets, packageSnippets, userSnippets]) {
+              for (const filepath in snippetSet) {
+                const snippetsBySelector = snippetSet[filepath]
+                this.add(filepath, snippetsBySelector)
+              }
+            }
+          })
+          this.doneLoading()
+        })
+      })
+    })
+  },
+
+  loadBundledSnippets (callback) {
+    const bundledSnippetsPath = CSON.resolve(path.join(getPackageRoot(), 'lib', 'snippets'))
+    this.loadSnippetsFile(bundledSnippetsPath, snippets => {
+      const snippetsByPath = {}
+      snippetsByPath[bundledSnippetsPath] = snippets
+      callback(snippetsByPath)
+    })
+  },
+
+  loadUserSnippets (callback) {
+    const userSnippetsPath = this.getUserSnippetsPath()
+    return fs.stat(userSnippetsPath, (error, stat) => {
```

Comment:
this `return` statement seems to not be needed
```suggestion
    fs.stat(userSnippetsPath, (error, stat) => {
```